homeassistant/util/distance.py

```diff
@@ -56,26 +85,66 @@ def __miles_to_meters(miles: float) -> float:
     return miles * 1609.344
 
 
+def __yards_to_meters(yards: float) -> float:
+    """Convert yards to meters."""
+    return yards * 0.9144
+
+
 def __feet_to_meters(feet: float) -> float:
     """Convert feet to meters."""
     return feet * 0.3048
 
 
+def __inches_to_meters(inches: float) -> float:
+    """Convert inches to meters."""
+    return inches * 0.0254
+
+
 def __kilometers_to_meters(kilometers: float) -> float:
     """Convert kilometers to meters."""
     return kilometers * 1000
 
 
+def __centimeters_to_meters(centimeters: float) -> float:
+    """Convert centimeters to meters."""
+    return centimeters * 0.01
+
+
+def __millimeters_to_meters(millimeters: float) -> float:
+    """Convert millimeters to meters."""
+    return millimeters * 0.001
+
+
 def __meters_to_miles(meters: float) -> float:
     """Convert meters to miles."""
     return meters * 0.000621371
 
 
+def __meters_to_yards(meters: float) -> float:
+    """Convert meters to yards."""
+    return meters * 1.09361
+
+
 def __meters_to_feet(meters: float) -> float:
     """Convert meters to feet."""
     return meters * 3.28084
 
 
+def __meters_to_inches(meters: float) -> float:
+    """Convert meters to inches."""
+    return meters * 39.3701
+
+
 def __meters_to_kilometers(meters: float) -> float:
     """Convert meters to kilometers."""
     return meters * 0.001
+
+
+def __meters_to_centimeters(meters: float) -> float:
+    """Convert meters to centimeters."""
+    return meters * 100
+
+
+def __meters_to_millimeters(meters: float) -> float:
+    """Convert meters to millimeters."""
+    return meters * 1000
```

Comment:
Please add tests

Reply:
Done in https://github.com/home-assistant/core/pull/40181/commits/19581142bf02aeaeda03390b9b1139268131f7fe